confmap/provider/httpprovider/provider.go

```diff
@@ -16,38 +16,87 @@ package httpprovider // import "go.opentelemetry.io/collector/confmap/provider/h
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
 	"io"
 	"net/http"
+	"os"
+	"path/filepath"
 	"strings"
 
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/confmap/provider/internal"
 )
 
+type TransportType string
+
 const (
-	schemeName = "http"
+	PlainText TransportType = "http"
+	TLS       TransportType = "https"
 )
 
 type provider struct {
-	client http.Client
+	client    http.Client
+	transport TransportType
 }
 
-// New returns a new confmap.Provider that reads the configuration from a file.
-//
-// This Provider supports "http" scheme, and can be called with a "uri" that follows:
+// New returns a new confmap.Provider that reads the configuration from a file using the configured transport mechanism.
+// There are two types of transport supported: PlainText (HTTP) and TLS (HTTPS).
 //
-// One example for http-uri be like: http://localhost:3333/getConfig
+// This Provider supports "http" and "https" schemes based on the transport chosen, and can be called with a "uri" that
+// follows:
 //
-// Examples:
-// `http://localhost:3333/getConfig` - (unix, windows)
-func New() confmap.Provider {
-	return &provider{client: http.Client{}}
+// One example for http-uri: http://localhost:3333/getConfig
+// One example for https-uri: https://localhost:3333/getConfig
+func New(transport TransportType) confmap.Provider {
```

Comment:
I worry somewhat about having a non-niladic initializer.  None of the other providers (that I'm aware of) require arguments to their initializer and having arguments there may complicate any attempt to expand the collector builder to support specifying `confmap.Provider`s to be included in a build.

Reply:
The latest version uses niladic initializers for the http and https config map providers.